### PR TITLE
feat: Add get lists endpoint

### LINKED ISSUE #85: Create the get lists endpoint
To complete this task, create a paginated endpoint where an authenticated user can get their lists. From the lists, it will be necessary to retrieve their id and name so they can be eventually used in the UI.

### PR DESCRIPTION
**Pull request by kevinszuchet**
Closes #85

### CODE REVIEW

File: test/unit/lists-adapters.spec.ts [hunk 1]
```diff
@@ -83,3 +90,62 @@ describe('when transforming a DB retrieved pick to a pick', () => {
...
+  describe('and there are not DB retrieved lists', () => {
+    beforeEach(() => {
+      dbGetLists = []
+      listsWithCount = {
+        lists: [],
+        count: 0
+      }
+    })
+
+    it('should return an empty array of picks and the count property as 0', () => {
```
Comment (LautaroPetaccio) [inline test/unit/lists-adapters.spec.ts:107]:
```suggestion
    it('should return an empty array of lists and the count property as 0', () => {
```

File: test/unit/lists-adapters.spec.ts [hunk 2]
```diff
@@ -83,3 +90,62 @@ describe('when transforming a DB retrieved pick to a pick', () => {
...
+      expect(fromDBGetListsToListsWithCount(dbGetLists)).toEqual(listsWithCount)
+    })
+  })
+
+  describe('and there are multiple DB retrieved picks', () => {
```
Comment (LautaroPetaccio) [inline test/unit/lists-adapters.spec.ts:112]:
```suggestion
  describe('and there are multiple DB retrieved lists', () => {
```

File: src/ports/lists/component.ts [hunk 3]
```diff
@@ -106,5 +106,18 @@ export function createListsComponent(
     }
   }
 
-  return { getPicksByListId, addPickToList, deletePickInList }
+  async function getLists(params: GetAuthenticatedAndPaginatedParameters): Promise<DBGetListsWithCount[]> {
+    const { userAddress, limit, offset } = params
+    // TODO: do we want to sort the lists using another criteria?
```
Comment (LautaroPetaccio) [inline src/ports/lists/component.ts:111]: That's a great question, I would sort it by `is_default_list` and then by date of creation. What do you think about that? Eventually, with new features in mind, we can add even more orders (list with the most recent added item, etc).

**Review by LautaroPetaccio — APPROVED** [review]
Looks great! I've just left some minor comments ✅

File: src/ports/lists/component.ts (hunk 3 above)
Comment (kevinszuchet) [reply src/ports/lists/component.ts:111]: I created a new issue #90 to tackle this. WDYT?

**Pull request merged**